h/search/core.py

```diff
@@ -34,14 +35,25 @@ class Search(object):
     :type stats: statsd.client.StatsClient
     """
     def __init__(self, request, separate_replies=False, stats=None, _replies_limit=200):
-        self.request = request
-        self.es = request.es
+        self.es_index = request.es.index
         self.separate_replies = separate_replies
         self.stats = stats
         self._replies_limit = _replies_limit
-
-        self.builder = self._default_querybuilder(request, self.es)
-        self.reply_builder = self._default_querybuilder(request, self.es)
+        # Order matters!!! The KeyValueMatcher must be run last,
+        # after all other qualifiers have popped off the params.
```

Comment:
We could but I'm not a huge fan of that 1) because it means the clear method doesn't actually clear everything and 2) it hides things inside the run method. I really like having this list here because it exposes the previously hidden complexity and I know that these are all the things it runs. (Minus the RepliesFilter which I'm ok with cause it's specific to that's how we get replies and the separate_replies feature.) The more we can make things explicit the more readable the code is. For example I didn't even know we had a KeyValueMatcher catch-all until I ran the old_search_tests.